ta4j-core/src/main/java/org/ta4j/core/trading/rules/AbstractRule.java

```diff
@@ -43,6 +43,8 @@ this software and associated documentation files (the "Software"), to deal in
      * @param isSatisfied true if the rule is satisfied, false otherwise
      */
     protected void traceIsSatisfied(int index, boolean isSatisfied) {
-        log.trace("{}#isSatisfied({}): {}", className, index, isSatisfied);
+        if ( log.isTraceEnabled() ) {
+            log.trace("{}#isSatisfied({}): {}", getClass().getSimpleName(), index, isSatisfied);
```

Comment:
should be `className` instead of `getClass().getSimpleName()`. Then we should not need an additional `isTraceEnabled()` check